### ISSUE TITLE
[Bug]: haptics enabled on images with option turned off 

### ISSUE THREAD

**Issue by thadvuolo** [Memmy-App/memmy#1103]
### Check for open issues

- [ ] I have verified that another issue for this is not open, or it has been closed and has not been fixed.

### Minimal reproducible example

Open an image from your feed. Zoom in and scroll around. Every touch has haptic feedback even when the option is disabled in options. 

### Expected Behavior

No haptics on images unless the option is enabled 

### Version

Latest

### App Store or TestFlight?

App Store

### iOS Version

17.x

### Device

iPhone 12 Max

**Comment by lattaai5** [comment]
Hello, I tried to fix the issue

This is what I did:

Modify the ViewerImageWrapper component to check if haptics are enabled before allowing touch feedback on image interactions. This will prevent haptic feedback when the option is disabled in settings.

I also created pull request: https://github.com/Memmy-App/memmy/pull/1104


> [!CAUTION]
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!
>
> The fix provided by Latta AI might not be complete and it can serve as an inspiration.

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt fixing issues on your repository, you can block this account.
